### PR TITLE
Dialect api

### PR DESCRIPTION
**Pull request by VitGottwald**

### CODE REVIEW

File: clients/cobol-dialect-api/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  jar: vscode.Uri;
+  snippets: vscode.Uri;
+}
+const mainExtensionId = "BroadcomMFD.cobol-language-support";
+
+interface unregister {
+  (): void;
+}
+
+export const registerDialect = async (
```
Comment (ishche) [inline clients/cobol-dialect-api/src/index.ts:16]: I think it makes sense to add a simple test for this.

File: clients/.vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2022 Broadcom.
```
Comment (ishche) [inline clients/.vscode/extensions.json]: it's 2023 already :)

File: clients/daco-dialect-support/.vscodeignore [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Ignore everything
```
Comment (ishche) [inline clients/daco-dialect-support/.vscodeignore:15]: copyright header is missing

File: clients/cobol-dialect-api/src/index.ts (hunk 1 above)
Comment (VitGottwald) [reply clients/cobol-dialect-api/src/index.ts:16]: I will do this in next iteration that will include refactoring of the API exported by COBOL LS extension. The focus of this PR is on the clean up of the API faced by dialect extender.

**Review by amanPras — APPROVED** [review]

**Review by ishche — APPROVED** [review]

**Review by Nurkambay — APPROVED** [review]

**Pull request merged**